x-pack/plugins/siem/server/lib/detection_engine/routes/rules/import_rules_route.ts

```diff
@@ -94,166 +94,165 @@ export const importRulesRoute = (router: IRouter, config: ConfigType) => {
             batchParseObjects.reduce<Array<Promise<ImportRuleResponse>>>((accum, parsedRule) => {
-              const importsWorkerPromise = new Promise<ImportRuleResponse>(
-                async (resolve, reject) => {
-                  if (parsedRule instanceof Error) {
-                    // If the JSON object had a validation or parse error then we return
-                    // early with the error and an (unknown) for the ruleId
+              const importsWorkerPromise = new Promise<ImportRuleResponse>(async resolve => {
+                if (parsedRule instanceof Error) {
+                  // If the JSON object had a validation or parse error then we return
+                  // early with the error and an (unknown) for the ruleId
+                  resolve(
+                    createBulkErrorObject({
+                      statusCode: 400,
+                      message: parsedRule.message,
+                    })
+                  );
+                  return null;
+                }
+                const {
+                  anomaly_threshold: anomalyThreshold,
+                  description,
+                  enabled,
+                  false_positives: falsePositives,
+                  from,
+                  immutable,
+                  query,
+                  language,
+                  machine_learning_job_id: machineLearningJobId,
+                  output_index: outputIndex,
+                  saved_id: savedId,
+                  meta,
+                  filters,
+                  rule_id: ruleId,
+                  index,
+                  interval,
+                  max_signals: maxSignals,
+                  risk_score: riskScore,
+                  name,
+                  severity,
+                  tags,
+                  threat,
+                  to,
+                  type,
+                  references,
+                  note,
+                  timeline_id: timelineId,
+                  timeline_title: timelineTitle,
+                  version,
+                  exceptions_list,
+                } = parsedRule;
+
+                try {
+                  validateLicenseForRuleType({
+                    license: context.licensing.license,
+                    ruleType: type,
+                  });
+
+                  const signalsIndex = siemClient.getSignalsIndex();
+                  const indexExists = await getIndexExists(
+                    clusterClient.callAsCurrentUser,
+                    signalsIndex
+                  );
+                  if (!indexExists) {
                     resolve(
                       createBulkErrorObject({
-                        statusCode: 400,
-                        message: parsedRule.message,
+                        ruleId,
+                        statusCode: 409,
+                        message: `To create a rule, the index must exist first. Index ${signalsIndex} does not exist`,
                       })
                     );
```

Comment:
I'm wondering if we could move this section of code up above the while loop? I think doing that would save us from not grabbing the signals index name and checking for its existence on each iteration.

Reply:
Agreed, there's no loop-specific information in this check (besides adding the ruleId to the error), we should move it outside and just exit early.

Reply:
Ok, yeah, it can be moved up. I will do that before merge it in.